scipy/stats/tests/test_hypotests.py

```diff
@@ -159,3 +160,79 @@ def test_callable_cdf(self):
+        y = np.arange(5)
+        assert_raises(ValueError, cramervonmises_2samp, x, y)
+        assert_raises(ValueError, cramervonmises_2samp, y, x)
+        assert_raises(ValueError, cramervonmises_2samp, [], y)
+        assert_raises(ValueError, cramervonmises_2samp, y, [1])
+        assert_raises(ValueError, cramervonmises_2samp, y, y, 'wrong_method')
+
+    def test_list_input(self):
+        x = [2, 3, 4, 7, 6]
+        y = [0.2, 0.7, 12, 18]
+        r1 = cramervonmises_2samp(x, y)
+        r2 = cramervonmises_2samp(np.array(x), np.array(y))
+        assert_equal((r1.statistic, r1.pvalue), (r2.statistic, r2.pvalue))
+
+    def test_example_conover(self):
+        # Example 2 in Section 6.2 of W.J. Conover: Practical Nonparametric
+        # Statistics, 1971.
+        x = [7.6, 8.4, 8.6, 8.7, 9.3, 9.9, 10.1, 10.6, 11.2]
+        y = [5.2, 5.7, 5.9, 6.5, 6.8, 8.2, 9.1, 9.8, 10.8, 11.3, 11.5, 12.3, 12.5, 13.4, 14.6]
+        r = cramervonmises_2samp(x, y)
+        assert_almost_equal(r.statistic, 0.262, decimal=3)
+        assert_almost_equal(r.pvalue, 0.18, decimal=2)
+
+    def test_exact_pvalue(self):
+        # the exact values are taken from Anderson: On the distribution of the
+        # two-sample Cramer-von-Mises criterion, 1962.
+        # The values are taken from Table 2, 3 and 4
+        p1 = _pval_cvm_2samp_exact(710, 5, 6)
+        p2 = _pval_cvm_2samp_exact(1897, 7, 7)
+        p3 = _pval_cvm_2samp_exact(576, 4, 6)
+        assert_almost_equal(p1, 0.103896, decimal=6)
+        assert_almost_equal(p2, 0.068182, decimal=6)
+        assert_almost_equal(p3, 0.009524, decimal=6)
+
+    def test_large_sample(self):
+        # for large samples, the statistic U gets very large
+        # do a sanity check that p-value is not 0, 1 or nan
+        np.random.seed(4367)
+        x = distributions.norm.rvs(size=1000000)
+        y = distributions.norm.rvs(size=900000)
+        r = cramervonmises_2samp(x, y)
+        assert_(0 < r.pvalue < 1)
+        r = cramervonmises_2samp(x, y+0.1)
+        assert_(0 < r.pvalue < 1)
+
+    def test_exact_vs_asymp(self):
+        x = np.arange(7)
+        y = x + 0.5
+        r1 = cramervonmises_2samp(x, y, method='exact')
+        r2 = cramervonmises_2samp(x, y, method='asymp')
+        assert_equal(r1.statistic, r2.statistic)
+        assert_almost_equal(r1.pvalue, r2.pvalue, decimal=2)
+
+    def test_method_auto(self):
+        x = np.arange(10)
+        y = x + 0.5
+        r1 = cramervonmises_2samp(x, y, method='exact')
+        r2 = cramervonmises_2samp(x, y, method='auto')
+        assert_equal(r1.pvalue, r2.pvalue)
+        # switch to asymp if one sample has more than 10 observations
+        x = np.arange(11)
+        r1 = cramervonmises_2samp(x, y, method='asymp')
+        r2 = cramervonmises_2samp(x, y, method='auto')
+        assert_equal(r1.pvalue, r2.pvalue)
+
+    def test_same_input(self):
+        # make sure trivial edge case can be handled
+        # note that _cdf_cvm_inf(0) = nan
```

Comment:
Do you mean that for this test `_cdf_cvm_inf` returns `nan`? If so, how is that turning into the correct result here? (I don't see how the NaN goes away.)

Reply:
The implementation returns a p-value of 1.0 in that case (since `tn=0 < 0.02`), so there is no NaN. I just wanted to point out that this edge case needs to be taken care of. I will update the test description

Reply:
I tried to make the comment more clear